tests/integration/test_storage_s3/test.py

```diff
@@ -685,6 +684,56 @@ def add_tales(start, end):
     assert run_query(instance, query).splitlines() == ["1001\t1001\t1001\t1001"]
 
 
+def test_s3_trivial_glob_should_not_list(started_cluster):
+    bucket = started_cluster.minio_bucket
+    instance = started_cluster.instances["dummy"]  # type: ClickHouseInstance
+    table_format = "column1 UInt32, column2 UInt32, column3 UInt32"
+    values = "(1, 1, 1)"
+    nights_per_job = 1001 // 30
+    jobs = []
+    for night in range(0, 1001, nights_per_job):
+
+        def add_tales(start, end):
+            for i in range(start, end):
+                path = "night_{}/tale.csv".format(i)
+                query = "insert into table function s3('http://{}:{}/{}/{}', 'CSV', '{}') values {}".format(
+                    started_cluster.minio_ip,
+                    MINIO_INTERNAL_PORT,
+                    bucket,
+                    path,
+                    table_format,
+                    values,
+                )
+                run_query(instance, query)
+
+        jobs.append(
+            threading.Thread(
+                target=add_tales, args=(night, min(night + nights_per_job, 1001))
+            )
+        )
+        jobs[-1].start()
```

Comment:
"I'm curious about the choice to use multithreading here - is it primarily to help the test run faster, or were there other factors that influenced this decision? I'd love to understand the reasoning behind it

Reply:
I copy-pasted a test above. I agree we don't need multithreading loading here.